api/kroxylicious-api/src/main/java/io/kroxylicious/proxy/service/BaseContributor.java

```diff
@@ -16,24 +16,30 @@
 
 /**
  * A convenience base class for creating concrete contributor subclasses using a typesafe builder
+ *
+ * @param <T> the service type
  */
-public abstract class BaseContributor<L> implements Contributor<L> {
+public abstract class BaseContributor<T> implements Contributor<T> {
 
-    private final Map<String, InstanceBuilder<? extends BaseConfig, L>> shortNameToInstanceBuilder;
+    private final Map<String, InstanceBuilder<? extends BaseConfig, T>> shortNameToInstanceBuilder;
 
-    public BaseContributor(BaseContributorBuilder<L> builder) {
+    /**
+     * Constructs the contributor using the supplied {@code builder}.
+     * @param builder builder
+     */
+    public BaseContributor(BaseContributorBuilder<T> builder) {
         shortNameToInstanceBuilder = builder.build();
```

Comment:
I know its not part of this PR, but `shortNameToInstanceBuilder = builder.build();` surprised me. The builder is building a map not a thing?

Reply:
Agreed, the naming is off.  `BaseContributorBuilder`=> `ContributorRegistryBuilder`.  ContributorRegistryBuilder should build a `ContributorRegistry` which might be composed of a Map.  Obviously not part of this PR.  @robobario